### PR TITLE
WC2-226 Integrate WFP CIAM as am auth provider

### PR DESCRIPTION
**Pull request by olethanh**
Allow login via WFP auth server. see the README.md.

Will re-conciliate user on email address, it will automatically create an user account if there is none present (with the username as email). will send an email to warn a new account has been created.

It still missing a special page if the user has no permission at all, I'm on it but I'm already opening this PR

Related JIRA tickets : WC2-226  & WC2-228 & WC2-227

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [na] Are my typescript files well typed
- [] New translations have been added or updated if new strings have been introduced in the frontend
- [na] My migrations file are included
- [NO] Are there enough tests
- [x] Documentation has been included (for new feature)

## Changes

- upgraded django-allauth to a version that support the PKCE challenge
- Move the whole code in a separate django app.
- Removed the old code
- see `plugins/wfp_auth/README.md`

## How to test

To test in local:

set the environment variable
-       WFP_AUTH_CLIENT_ID: Check in one password
-       WFP_AUTH_ACCOUNT: Which Iaso account user should be associated to
-       WFP_EMAIL_RECIPIENTS_NEW_ACCOUNT: email that should receive the notification for a new user.

Try login via the CIAM button (it's only added if the above has been configured). Create an account on the WFP server, validate the e-mail. You can try login before the email is not validated, it should fail.

## Print screen / video

https://github.com/BLSQ/iaso/assets/82500/caa047b2-b243-46d6-9473-ca000a6c2d0e

Note that the front end button has not been translated so far.  
The email haven't been translated either but I'm not sure this make sense

### CODE REVIEW

**Comment by olethanh** [comment]
updated the PR  with a proper error message (if you can take a look @beygorghor ) and a better email text

image

File: plugins/wfp_auth/templates/wfp_auth/login_subtemplate.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<form method="post" action="{% url 'wfp_login' %}" type="submit"  class="auth__form">
+  {% csrf_token %}
+  <button type="submit" class="button">Login via WFP CIAM</button>
```
Comment (beygorghor) [inline plugins/wfp_auth/templates/wfp_auth/login_subtemplate.html]: This is not translated

File: hat/assets/js/apps/Iaso/components/errors/PageNoPerms.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Paper,
+    Typography,
+} from '@material-ui/core';
+import Info from '@material-ui/icons/Info';
+import { ClassNameMap } from 'notistack';
+import TopBar from '../nav/TopBarComponent';
+
+import MESSAGES from './messages';
+
+// @ts-ignore
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/components/errors/PageNoPerms.tsx:17]: you can remove this ignore and set fontWeight to "lighter" to fix the warning

**Review by beygorghor — COMMENTED** [review]
I'm just wondering if we can manage ti style a bit this page:
Screenshot 2023-05-30 at 11 23 53
I had it while trying to configure the feature, i arrived on this page : http://localhost:8081/accounts/wfp/login/callback/?error_description=PKCE+is+mandatory+for+this+application.+PKCE+Challenge+is+not+provided+or+is+not+upto+RFC+7636+specification.&error=invalid_request

**Review by mathvdh — APPROVED** [review]

**Pull request merged**